SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialStatics.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialStatics.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "GeneralProjectSettings.h"
+#include "SpatialConstants.h"
+#include "SpatialGDKSettings.h"
+#include "Utils/ActorGroupManager.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialStatics);
+
+bool USpatialStatics::SpatialNetworkingEnabled()
+{
+    return GetDefault<UGeneralProjectSettings>()->bSpatialNetworking;
+}
+
+bool USpatialStatics::SpatialOffloadingEnabled()
+{
+    return SpatialNetworkingEnabled() && GetDefault<USpatialGDKSettings>()->bEnableOffloading;
+}
+
+bool USpatialStatics::IsAuthoritativeWorkerType(const AActor* Actor)
+{
+    if (Actor == nullptr)
+    {
+        UE_LOG(LogSpatialStatics, Warning, TEXT("Actor was nullptr in USpatialStatics::IsAuthoritativeWorkerType"));
+        return false;
+    }
+
+    if (!SpatialNetworkingEnabled() || !SpatialOffloadingEnabled()) 
+    {
+        return Actor->HasAuthority();
+    }
+
+    UWorld* World = Actor->GetWorld();
+    if (!World)
+    {
+        UE_LOG(LogSpatialStatics, Warning, TEXT("World was nullptr in USpatialStatics::IsAuthoritativeWorkerType for actor: %s"), *Actor->GetName());
```

Comment:
Why warning? Is there a case where this can happen where a user didn't call it somewhere it wasn't supposed to be called?

Reply:
what would you suggest, a `check(World)` ?